### DISCUSSION TITLE
SelectBySingleFeaturePerformance - Why numerical features only?

### DISCUSSION THREAD

**Discussion by Fernakamuta** [feature-engine/feature_engine#671] [Q&A]
Hello, 
I tried to used `SelectBySingleFeaturePerformance` to see the roc-auc score of some categorical features with an univariate lgbm model.
Since lgbm accepts categorical features as pandas category dtypes, I was using this way... But I am not able to use them in current  SelectBySingleFeaturePerformance implementation because it uses the [find_or_check_numerical_variables](https://github.com/feature-engine/feature_engine/blob/dade0f0df0158e5d53cad0db7aa5c4007e299415/feature_engine/variable_handling/variable_type_selection.py#L19).

Maybe we could replace it by `find_categorical_and_numerical_variables`?

**Comment by solegalli** [comment]
Hey @Fernakamuta 

Interesting point. Thank you for the suggestion.

It is a hard one. If we allow the transformer to accept categorical variables, it is going to fail every time the user passes an sklearn classifier or regressor. So we will be failing on the great majority of algorithm implementations so that it can accept lightGBMs.

Would it not be possible to use ordinal encoding on the variables before passing it to the lightGBM instead? that should work.
